### PR TITLE
[new] suse support for rhn_register

### PR DESCRIPTION
**Pull request by jgrumboe**

### CODE REVIEW

**Comment by jgrumboe** [comment]
@LalatenduMohanty

**Comment by jgrumboe** [comment]
may someone have a look at this PR?

**Comment by pvalena** [comment]
Hi, i will test this soon. Can you please provide 'Configuration' and 'Options Reference' in README.md?

**Comment by jgrumboe** [comment]
Since Suse Manager (< 3.0) relies also on Spacewalk like Redhat Satellite (< 6.0) does there are no other options/parameters in the usage of rhn_register. So i only reflected the suse support in the comments. Hope this is ok.

**Comment by jgrumboe** [comment]
Hi, when do you think this PR will be merged?

**Comment by jgrumboe** [comment]
Can somebody merge this, please? (possibly @hferentschik )

**Comment by hferentschik** [comment]
So this is really based on pull request #68, right? It gets a bit confusing. Again, what is the issue you are addressing here. If there is none please create it and make sure the commit(s) refer to it.

Maybe you want to close these two pull requests and sort things out on a new branch and create a combined pull request?

**Comment by LalatenduMohanty** [comment]
@jgrumboe Can you please point us to a SUSE Vagrantbox, so that we can test the PR?

File: README.md [hunk 1]
```diff
@@ -23,7 +23,7 @@ If you are on Fedora, you can install the packaged version of the plugin by runn
 
 The plugin is designed in an registration-manager-agnostic way which means that plugin itself does not depend on any OS nor way of registration. vagrant-registration only calls registration capabilities for given guest, passes the configuration options to them and handles interactive registration.
 
-That being said, this plugin currently ships only with registration capability files for RHEL's Subscription Manager and `rhn_register`. Feel free to submit others.
+That being said, this plugin currently ships only with registration capability files for RHEL's Subscription Manager and `rhn_register` which is used by Redhat Satellite and Suse Manager. Feel free to submit others.
```
Comment (strzibny) [inline README.md]: Spelling is "Red Hat Satellite"

**Comment by hferentschik** [comment]
> Actually I suggested @jgrumboe to separate the two PRs as they address two different issues.

I am not saying to create a single pull request with a single commit. If there are two issues there should be two commits. However, if the two fixes are based on each other, then I am fine to have a single pull request and reviewing both in conjunction. 

Provided the two fixes are actually unrelated, the pull requests should be made from independent feature branches of master. That's the best option.

Either way, the pull request should be re-crafted to address the missing issue keys and to improve the commit message.

> This PR is for adding SUSE support so that vagrant-registration can be used for SUSE guest registration. 

Can we actually test this? Provided we can download a SUSE box we should be able to test this via Cucumber, right?

**Comment by hferentschik** [comment]
@jgrumboe can you re-work the pull request to address the given feedback?

**Comment by jgrumboe** [comment]
Hi @hferentschik,
this request (#67) is not related to #68 and they should not be based on each other. I started both as separate branches from the master. (How do you think it's based on #68, do i overlook something?)

I'm not familiar with cucumber tests and i also don't know if there's any suse box available for download (probably at Suse Studio).
So I'm a little bit stuck here and don't know what to do now - just wanted to contribute and my changes work for me here, since we have suse manager and red hat satellite both in place here.
@hferentschik what rework do you actually want from me?

@strzibny: I will correct the spelling, thx.

greetings, Jo

**Comment by hferentschik** [comment]
> this request (#67) is not related to #68 and they should not be based on each other. I started both as separate branches from the master. (How do you think it's based on #68, do i overlook something?)

Looking at the commits of the pull request - https://github.com/projectatomic/adb-vagrant-registration/pull/67/commits - it used to be three commits in the pull request, two of which were from the other pull request. Now there are five commits which also seems wrong. I assume there is something wrong with your workflow. Have a look at the comment to my other pull request to see an example workflow. Right now I am not quite sure what to make of these commits.

> I'm not familiar with cucumber tests and i also don't know if there's any suse box available for download 

How do you test it then? Where is your Suse coming from?

> just wanted to contribute and my changes work for me here, since we have suse manager and red hat satellite both in place here.

Sure, but it's a bit hard for us to merge this w/o a test to show the behavior works or at least some steps to reproduce what this change is doing. At the very least we need the latter and I guess to verify even any manual steps we need a way to get hold of a SUSE VM. Surely there must be some, since you are using them.

**Comment by jgrumboe** [comment]
thx, as also commented in the other workflow i will start over again with a new issue entry.

Regarding "my" suse vm: i can use it and update it since we've paying a support contract for it.
btw, as long as you are only interested in installing a vm in virtualbox (as long as you don't care about support/patches) you can register for an eval download at suse.com and download the iso images.
I'm not quite sure how to test rhn_register against a suse manager 2.x afterwards.
How are you testing rhn_register against a red hat satellite 5.x now?

**Comment by hferentschik** [comment]
>  @i will start over again with a new issue entry.

Awesome thanks. Much appreciated. We can take one thing at a time. The other pull request seems quite trivial and we should be able to get this in quickly.

> Regarding "my" suse vm: i can use it and update it since we've paying a support contract for it.
btw, as long as you are only interested in installing a vm in virtualbox (as long as you don't care about support/patches) you can register for an eval download at suse.com and download the iso images.

Being able to just verify the patch against a manual download is a good first step. We can see whether we can figure something else out later.

**Comment by hferentschik** [comment]
Closing this pull request for now awaiting an update.

**Pull request closed without merging**